src/gcovr/merging.py

```diff
@@ -382,15 +411,40 @@ def merge_function(
       - ``options.func_opts.merge_function_use_line_max``
       - ``options.func_opts.separate_function``
     """
-    if left.demangled_name != right.demangled_name:
-        raise AssertionError("Function demangled name must be equal.")
-    if left.name != right.name:
-        raise AssertionError("Function name must be equal.")
+    if left.demangled_name is not None and right.demangled_name is not None:
+        if (
+            not options.func_name_opts.ignore_function_name
+            and left.demangled_name != right.demangled_name
+        ):
+            raise AssertionError(
+                f"Demangled name must be equal, got {left.demangled_name} and {right.demangled_name}."
```

Comment:
Demangled function names ...

Should these exceptions also point to the relevant option to use to fix things?